internal/controller/stats_reporter.go

```diff
@@ -0,0 +1,144 @@
+package controller
```

Comment:
Don't centralize metrics registering.

The common Kubernetes pattern in metrics for controllers is to delegate registering them to each controller package, like this for example:
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/certificates/clustertrustbundlepublisher/metrics.go#L73-L78

You may want to implement your own `Register()` function in `pkg/metrics`.

I'd suggest you experimented with `"k8s.io/component-base/metrics"` to see if it works well for the controller-runtime metrics registry.